ledger.go

```diff
@@ -200,29 +210,122 @@ func (l *Ledger) Addresses(hrp string, addresses int) ([]*Address, error) {
 // SignHash attempts to sign the [hash] with the provided path [addresses].
 // [addresses] are appened to the [pathPrefix] (m/44'/9000'/0'/0).
 func (l *Ledger) SignHash(hash []byte, addresses []uint32) ([][]byte, error) {
-	msgHash := []byte{
-		CLA,
-		INSSignHash,
-		0x0,
-		0x0,
-	}
 	pathBytes, err := bip32bytes(pathPrefix, 3)
 	if err != nil {
 		return nil, err
 	}
 	data := []byte{byte(len(addresses))}
 	data = append(data, hash...)
 	data = append(data, pathBytes...)
-	msgHash = append(msgHash, byte(len(data)))
-	msgHash = append(msgHash, data...)
+	data = append([]byte{byte(len(data))}, data...)
```

Comment:
To me this pattern is unfamiliar, I suggest a helper function so the code reads like:
`data = prependLength(data)`

Also what happens if `len(data)` does not fit in a byte?

Reply:
The length should never exceed 73 bytes as the hash will be 32 and the pathBytes can only be 41 bytes max, as there is a check in crypto.go:30